src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -62,26 +82,26 @@
      * @param argumentValues   the map of arguments
      * @param path             the logical path to the field in question
      * @param e                the exception that occurred
+     *
+     * @deprecated implement the new {@link DataFetcherExceptionHandler} interface and pass it in as a parameter to the strategy
      */
-    @SuppressWarnings("unused")
+    @SuppressWarnings({"unused", "DeprecatedIsStillUsed"})
+    @Deprecated
     protected void handleDataFetchingException(
             ExecutionContext executionContext,
             GraphQLFieldDefinition fieldDef,
             Map<String, Object> argumentValues,
             ExecutionPath path,
             Exception e) {
-        ExceptionWhileDataFetching error = new ExceptionWhileDataFetching(path, e);
-        executionContext.addError(error);
```

Comment:
this will be a breaking change to 3rd-party implementations of `ExecutionStrategy` which _don't_ override `handleDataFetchingException` but depend on it to add errors to the `ExecutionContext`. Can we call `SimpleDataFetcherExceptionHandler#accept(DataFetcherExceptionHandlerParameters)` here?